### PR TITLE
Add Cypress ESLint Plugin and ESLint README instructions

### PR DESCRIPTION
**Pull request by avivbenchorin**
### Description
Currently, the repository has ESLint installed, but has neither set up Cypress-specific rules nor provided instructions on how to use ESLint. This PR adds the Cypress ESLint Plugin to the ESLint configurations to provide, as well as updating the README to include instructions on installing and running ESLint. It also changes the global indentation rule to be 2 spaces instead of 4 and applies the new rules to all files.

Signed-off-by: Aviv Benchorin <benchori@amazon.com>

### Issues Resolved

N/A

### Check List

- [ ] New functionality includes testing. 
    - [x] All tests pass
- [ ] New functionality has been documented. 
    - [ ] New functionality has javadoc added 
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by seraphjiang — APPROVED** [review]
nice enhancement to development process to ensure quality.
LGTM

**Review by tianleh — APPROVED** [review]

File: cypress/integration/demo_tests/demo_spec.js [hunk 1]
```diff
@@ -4,121 +4,121 @@
...
-        cy.get('[data-test-subj="querySubmitButton"]').click()
+    // Submit a search query
+    cy.get('[data-test-subj="queryInput"]').type("products.taxless_price >= 60 AND category : Women's Clothing")
+    cy.get('[data-test-subj="querySubmitButton"]').click()
 
-        // Select the "category" field
-        /**
+    // Select the "category" field
+    /**
          * Issue: want to be able to "hover" over the category list element,
```
Comment (kavilla) [inline cypress/integration/demo_tests/demo_spec.js]: NIT: spacing here looks a little off

**Review by kavilla — APPROVED** [review]

**Comment by kavilla** [comment]
Another nitpick but might easier to have a little bit more descriptive branch names. Might get hard to track if it gets into the double digits.

File: cypress/integration/demo_tests/demo_spec.js (hunk 1 above)
Comment (avivbenchorin) [reply cypress/integration/demo_tests/demo_spec.js]: Should be fixed now!

**Comment by AvivBenchorin** [comment]
> Another nitpick but might easier to have a little bit more descriptive branch names. Might get hard to track if it gets into the double digits

**Pull request merged**